examples/cluster/plot_segmentation_toy.py

```diff
@@ -30,8 +30,10 @@
 #           Gael Varoquaux <gael.varoquaux@normalesup.org>
 # License: BSD 3 clause
 
+# %%
+# Generate the data
+# -----------------
 import numpy as np
-import matplotlib.pyplot as plt
 
 from sklearn.feature_extraction import image
```

Comment:
It appears to me that this import is first used on line 74, thus I would suggest:
```diff
-38 from sklearn.feature_extraction import image
-74 graph = image.img_to_graph(img, mask=mask)

+74 from sklearn.feature_extraction import image
+75
+76 graph = image.img_to_graph(img, mask=mask)
```

I would also do this for `import matplotlib.pyplot as plt` and `from sklearn.cluster import spectral_clustering`:

```diff
-39 from sklearn.cluster import spectral_clustering
-59 import matplotlib.pyplot as plt
-84 labels = spectral_clustering(graph, n_clusters=4, eigen_solver="arpack")

+84 from sklearn.cluster import spectral_clustering
+85 import matplotlib.pyplot as plt
+86
+87 labels = spectral_clustering(graph, n_clusters=4, eigen_solver="arpack")
```